### ISSUE TITLE
Build failure with template-haskell-2.22 (GHC 9.10)

### ISSUE THREAD

**Issue by andreasabel** [tsurucapital/vector-th-unbox#15]
```
Data/Vector/Unboxed/Deriving.hs:72:32: error: [GHC-83865]
    • Couldn't match type ‘Pat’ with ‘ArgPat’
      Expected: [ArgPat]
        Actual: [Pat]
    • In the first argument of ‘Clause’, namely ‘pats’
      In the expression: Clause pats (NormalB body) []
      In the second argument of ‘FunD’, namely
        ‘[Clause pats (NormalB body) []]’
   |
72 |     method = FunD name [Clause pats (NormalB body) []]
   |                                ^^^^
```
A revision won't do it for GHC 9.10, this requires fix and release.

**Comment by andreasabel** [comment]
This problem might fix itself in the next pre-release of GHC 9.10:
- https://gitlab.haskell.org/ghc/ghc/-/commit/36a75b80ebe592f582f3f349e8c73b8293d49ed1

**Comment by Fuuzetsu** [comment]
9.10 seems to be building these days at least so maybe it's fixed https://github.com/tsurucapital/vector-th-unbox/actions/runs/12226203767/job/34101154748
